File: samples/bedrock-agent/lib/bedrock-guardrail-stack.ts
/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import * as cdk from 'aws-cdk-lib';
import {Construct} from 'constructs';
import { bedrock } from '@cdklabs/generative-ai-cdk-constructs';
import {  General, InformationTechnology, PiiEntitiesConfigAction, Topic } from '@cdklabs/generative-ai-cdk-constructs/lib/cdk-lib/bedrock';

export class BedrockGuardrailStack extends cdk.Stack {
    constructor(scope: Construct, id: string, props: cdk.StackProps) {
      super(scope, id, props);

    const guardrails = new bedrock.Guardrail(this,'bedrockGuardrails',{
        name: "my-BedrockGuardrails",
        description: "Legal ethical guardrails.",
    });

    guardrails.addSensitiveInformationPolicyConfig([{
      type: General.EMAIL,
      action:   PiiEntitiesConfigAction.BLOCK
    },
    {
        type: InformationTechnology.IP_ADDRESS,
        action:   PiiEntitiesConfigAction.BLOCK  
    }],{
        name: "CUSTOMER_ID", 
        description: "customer id",
        pattern: "/^[A-Z]{2}\d{6}$/",
        action: "BLOCK", 
    });
    // const customerIdRegex = /^[A-Z]{2}\d{6}$/;
    // console.log(customerIdRegex.test('AB123456')); // true
    // console.log(customerIdRegex.test('a123456')); // false

    const topic = new Topic(this,'topic');
    topic.financialAdviceTopic()
    topic.politicalAdviceTopic()
    
    guardrails.addTopicPolicyConfig(topic)

    guardrails.uploadWordPolicyFromFile('./scripts/wordsPolicy.csv')
    
}



}

File: samples/dotnet-samples/chatbot-demo/src/ChatbotDemo.Infrastructure/Stacks/BedrockGuardrailStack.cs
using Amazon.CDK;
using Amazon.CDK.AwsBedrock;
using Cdklabs.GenerativeAiCdkConstructs.Bedrock;
using Constructs;

namespace ChatbotDemo.Infrastructure.Stacks
{
    public class BedrockGuardrailStack : Stack
    {
        public string GuardrailId { get; set; }
        public string GuardrailVersion { get; set; }
        internal BedrockGuardrailStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
        {
            var guardrails = new Guardrail(this, $"{id}-guardrail", new GuardrailProps
            {
                Name = $"{id}-guardrail",
                Description = "Legal ethical guardrails",
            });

            guardrails.AddSensitiveInformationPolicyConfig(new ISensitiveInformationPolicyConfigProps[]
            {
                new SensitiveInformationPolicyConfigProps
                {
                    Type = General.EMAIL,
                    Action = PiiEntitiesConfigAction.BLOCK,
                },
                new SensitiveInformationPolicyConfigProps
                {
                    Type = InformationTechnology.IP_ADDRESS,
                    Action = PiiEntitiesConfigAction.BLOCK
                }
            }, new CfnGuardrail.RegexConfigProperty
            {
                Name = "CUSTOMER_ID",
                Description = "Customer Id",
                Pattern = "/^[A-Z]{2}\\d{6}$/",
                Action = "BLOCK",
            });

            var topicPolicy = new Topic(this, $"{id}-guardrail-topic");
            topicPolicy.FinancialAdviceTopic();
            topicPolicy.PoliticalAdviceTopic();
            
            guardrails.AddTopicPolicyConfig(topicPolicy);

            GuardrailId = guardrails.GuardrailId;
            GuardrailVersion = guardrails.GuardrailVersion;
            _ = new CfnOutput(this, "guardrailVersion", new CfnOutputProps
            {
                Value = guardrails.GuardrailVersion,
                Description = "Guardrail Version"
            });
            _ = new CfnOutput(this, "guardrailId", new CfnOutputProps
            {
                Value = guardrails.GuardrailId,
                Description = "Guardrail Id"
            });
        }
    }
}